### PR TITLE
Kulla

### PR DESCRIPTION
**Pull request by kudav**
Various changes, I'll start with feature branches here from now on as well:

beams3d_slow:
- reorganized function
- implemented new collision operators
- option to supply multiple ion species

beams3d_apply_bpert:
- improvements to calculation

beams3d_normdist:
- made version check meaningful

beams3d_from_fidasim:
- to be implemented

fieldlines_follow:
- check for file type (beams3d or fieldlines), and appropriate procedure

plot_fieldlines:
- added option for color coding the traced path lenght
- added keyboard movement

VMECplot:
-  now has STL export of equilibria
- option to show min/max values in 2D plots

isotoro:
- option for specifying file name

read_hdf5:
- fix for reading in subgroup attributes

plot_fidasim:
- Neutral density plots
- birth profiles
- 2D profiles
- supplying indices for toroidal and poloidal slices

get_bes_fida_aug_data.m:
- added option for plotting the background bremsstrahlung profile
- fixed instrument function

plot_fidasim_profiles:
- option for background bremsstrahlung
- specify plotting line style

New functions:
- beams3d_suv2rzp converts points (ie the beams3d distribution grid) to cylindrical coordinates
- beams3d_calc_beamdensity calculates the 3D beam density distribution - should be merged with beams3d_beamdensity

I need to clean up some things before merging, so this is a draft for now.

### CODE REVIEW

File: BEAMS3D/beams3d_calc_srhoxi.m
Comment (lazersos) [inline BEAMS3D/beams3d_calc_srhoxi.m:1]: Proper header text with description, date, version, author, and example usage required.

File: BEAMS3D/beams3d_from_fidasim.m
Comment (lazersos) [inline BEAMS3D/beams3d_from_fidasim.m:1]: Proper header text with description, date, version, author, and example usage required.

File: WALL/plot_wall_acc.m
Comment (lazersos) [inline WALL/plot_wall_acc.m:1]: This isn't the behavior we want, is it?

File: utilities/get_bes_fida_aug_data.m
Comment (lazersos) [inline utilities/get_bes_fida_aug_data.m:1]: AUG specific routines should be in a separate AUG repository like we do for W7-X.

**Review by lazersos — COMMENTED** [review]
There are a few points to clarify.  Also we should probably create a separate FIDASIM directory for FIDASIM routines.

**Review by kudav — COMMENTED** [review]
Addressed the points and cleaned up the superfluous/unfinished routines.

File: BEAMS3D/beams3d_suv2rzp.m
Comment (lazersos) [inline BEAMS3D/beams3d_suv2rzp.m:1]: Needs proper help header.

File: BEAMS3D/beams3d_slow.m
Comment (lazersos) [inline BEAMS3D/beams3d_slow.m:1]: Can you show that this routine returns simliar results to the old routine for the proper selection of log_type?

**Review by lazersos — CHANGES_REQUESTED** [review]
Still a couple more things to do.

File: BEAMS3D/beams3d_suv2rzp.m
Comment (lazersos) [reply BEAMS3D/beams3d_suv2rzp.m:1]: Still needs a 'proper' header like other files.

**Review by lazersos — CHANGES_REQUESTED** [review]
Still have to give a proper header and show that slow produces similar results.

**Comment by kudav** [comment]
Here is the heating plot from beams3d_slow with the current commit:
`slow=beams3d_slow(data,'NRL','plots');`

grafik

and from commit 18707fd7a328bd6a6156583a45ba8ad5bb9638ab (second), which was where we introduced the multiple coulomb logarithms:
`
slow=beams3d_slow(data,'NRLions_logarithm','plots');`

grafik

The calculation between the two did not change, thus the output is numerically the same.

File: BEAMS3D/beams3d_slow.m
Comment (lazersos) [reply BEAMS3D/beams3d_slow.m:1]: Confirmed.

**Review by lazersos — APPROVED** [review]
Merge. We will add your biological and technological distinctiveness to our own.

**Pull request merged**